pkg/controller/registry/resolver/cache/predicates.go

```diff
@@ -329,3 +330,41 @@ func (c countingPredicate) String() string {
 func CountingPredicate(p Predicate, n *int) Predicate {
 	return countingPredicate{p: p, n: n}
 }
+
+type evaluatorPredicate struct {
+	evaluator constraints.Evaluator
+	rule      string
+	message   string
+}
+
+func (ep *evaluatorPredicate) Test(entry *Entry) bool {
+	props := make([]map[string]interface{}, len(entry.Properties))
+	for i, p := range entry.Properties {
+		var v interface{}
+		if err := json.Unmarshal([]byte(p.Value), &v); err != nil {
+			continue
+		}
+		props[i] = map[string]interface{}{
+			"type":  p.Type,
+			"value": v,
+		}
+	}
+
+	ok, err := ep.evaluator.Evaluate(map[string]interface{}{"properties": props})
+	if err != nil {
+		return false
```

Comment:
same here: should we log or panic?

Reply:
In the original code, it was a panic. However, as you can see other predicates don't panic on json unmarshal error as we simply ignore it. So that's why I do it this way. We currently don't have logger for predicates so there is no logging for that either.

Reply:
How hard would it be to add logging? If we don't want to handle this in this PR, can we track this somewhere? No logging could lead to some hard to find bugs. Especially with a generic constraints evaluator, being in the dark here could be a bit painful.

Reply:
I'd even suggest adding the panic back until we have logging